ouroboros-consensus/src/Ouroboros/Storage/VolatileDB/Impl.hs

```diff
@@ -179,20 +182,22 @@ openDB :: (HasCallStack, IOLike m, Ord blockId, NoUnexpectedThunks blockId)
        -> ThrowCantCatch (VolatileDBError blockId) (STM m)
        -> Parser e m blockId
        -> Int
+       -> ResourceRegistry m
```

Comment:
This is mainly because of how VolatileDB is initialized. There is nothing that makes sure that closeDB is called, since `withDB` is not used, but `openDB` and `closeDB`. By creating a registry and passing it as an argument, this issue is fixed. I also think this approach fits well how the code already handles other resources.